### ISSUE TITLE
New Unit Testing Settings: Run Twice, Randomize Order

### ISSUE THREAD

**Issue by retailcoder** [rubberduck-vba/Rubberduck#5583]
**Justification**
Writing tests can be hard and complicated sometimes, and state may accidentally carry between executions; by running each test twice, random-sorted per test module (which would also be random-sorted per project), the test engine could potentially detect and report such problems. This could help our users write better unit tests for their code.

**Description**
When the state of a test is "unknown" and this setting is enabled, the test engine would execute each unit test twice and report each execution separately somehow (a tooltip is good enough for a v1), with the "overall" result for the test being the worst outcome.

**Additional context**
The modifications necessary to make this happen should make it easier to further tweak the test engine to, say, support executing a test multiple times with different parameter values given a `@TestCase` annotation...

**Comment by BZngr** [comment]
Really interesting idea/feature.  Wondering if in addition to 'random', offer 'reverse order'.  Simply running the default test sequence in reverse would simplify the user's troubleshooting efforts because the sequence is still deterministic and easily repeatable - my suspicion is that reversing the order would uncover most of the issues of 'state retention/leakage'.  

Further, if a randomized test sequence gets a different result than a deterministic sequence, it would seem the user may be wishing for a way to retain and re-run the failing randomized sequence.

**Comment by retailcoder** [comment]
Good point - it's pseudo-random though, so technically "repeat last run" could retain the RNG seed and re-the failed randomized tests... But then if there's shared state, nothing says repeating the last run will produce the same outcome: I think "repeat last run" needs to retain the execution order just in case, but yeah having the ability to run them in reverse would also be a useful configuration setting for this.
